openbas-api/src/main/java/io/openbas/service/ChallengeService.java

```diff
@@ -113,7 +119,16 @@ public ChallengesReader playerChallenges(String exerciseId, User user) {
                 injectExpectation -> {
                   Challenge challenge = injectExpectation.getChallenge();
                   challenge.setVirtualPublication(injectExpectation.getCreatedAt());
-                  return new ChallengeInformation(challenge, injectExpectation);
+                  InjectStatus injectStatus =
+                      injectExpectation.getInject().getStatus().orElseThrow();
+                  ChallengeAttemptId challengeAttemptId =
+                      challengeAttemptId(challenge.getId(), injectStatus.getId(), user.getId());
+                  ChallengeAttempt challengeAttempt =
+                      this.challengeAttemptService
+                          .challengeAttempt(challengeAttemptId)
+                          .orElse(challengeAttempt(challengeAttemptId));
```

Comment:
This is very confusing (same method name/same parameter but from two different classes)

Reply:
Is it less confusing now that the utils methods are renamed?

Reply:
Perfect for me, thx!